// terra-application/lib/terra-dev-site/test/application-navigation/ApplicationNavigationTest.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theme = require("terra-application/lib/theme");

var _applicationIntl = require("../../../application-intl");

var _applicationBase = _interopRequireDefault(require("../../../application-base"));

var _applicationNavigation = _interopRequireDefault(require("../../../application-navigation"));

var _navigationPrompt = _interopRequireDefault(require("../../../navigation-prompt"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var PendingAction = function PendingAction(_ref) {
  var index = _ref.index,
      onClick = _ref.onClick,
      navDisabled = _ref.navDisabled;
  return /*#__PURE__*/_react.default.createElement("p", null, "Toggle pending action", "".concat(index, ": "), /*#__PURE__*/_react.default.createElement("button", {
    id: "pending-action-toggle-".concat(index),
    type: "button",
    onClick: onClick
  }, navDisabled ? 'Disable' : 'Enable'));
};

PendingAction.propTypes = {
  index: _propTypes.default.string,
  onClick: _propTypes.default.func,
  navDisabled: _propTypes.default.bool
};

var PageContent = function PageContent(_ref2) {
  var title = _ref2.title;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      hasPendingAction1 = _useState2[0],
      setHasPendingAction1 = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      hasPendingAction2 = _useState4[0],
      setHasPendingAction2 = _useState4[1];

  return /*#__PURE__*/_react.default.createElement("div", {
    "data-nav-test-content": true
  }, /*#__PURE__*/_react.default.createElement("p", null, title), /*#__PURE__*/_react.default.createElement(PendingAction, {
    index: "1",
    navDisabled: hasPendingAction1,
    onClick: function onClick() {
      setHasPendingAction1(!hasPendingAction1);
    }
  }), /*#__PURE__*/_react.default.createElement(PendingAction, {
    index: "2",
    navDisabled: hasPendingAction2,
    onClick: function onClick() {
      setHasPendingAction2(!hasPendingAction2);
    }
  }), hasPendingAction1 ? /*#__PURE__*/_react.default.createElement(_navigationPrompt.default, {
    description: "Pending Action 1"
  }) : undefined, hasPendingAction2 ? /*#__PURE__*/_react.default.createElement(_navigationPrompt.default, {
    description: "Pending Action 2"
  }) : undefined);
};

PageContent.propTypes = {
  title: _propTypes.default.string
};

var ApplicationNavigationTest = function ApplicationNavigationTest() {
  var applicationIntl = (0, _react.useContext)(_applicationIntl.ApplicationIntlContext);

  var theme = _react.default.useContext(_theme.ThemeContext);

  var _useState5 = (0, _react.useState)('page_1'),
      _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
      activeNavItem = _useState6[0],
      setActiveNavItem = _useState6[1];

  var _useState7 = (0, _react.useState)(false),
      _useState8 = (0, _slicedToArray2.default)(_useState7, 2),
      loggedOut = _useState8[0],
      setLoggedOut = _useState8[1];

  var navigationItems = [{
    key: 'page_1',
    text: 'Page 1'
  }, {
    key: 'page_2',
    text: 'Page 2'
  }, {
    key: 'page_3',
    text: 'Page 3'
  }];
  return (
    /*#__PURE__*/
    // Disable prompt unloading since we're not testing that and don't want to prevent sessions ending in firefox if there are problems
    _react.default.createElement(_applicationBase.default, {
      unloadPromptIsDisabled: true,
      locale: applicationIntl.locale,
      themeName: theme.className
    }, loggedOut ? /*#__PURE__*/_react.default.createElement("p", null, "Logged Out") : /*#__PURE__*/_react.default.createElement(_applicationNavigation.default, {
      titleConfig: {
        title: 'ApplicationNavigation Test'
      },
      navigationItems: navigationItems,
      activeNavigationItemKey: activeNavItem,
      onSelectNavigationItem: function onSelectNavigationItem(key) {
        setActiveNavItem(key);
      },
      onSelectLogout: function onSelectLogout() {
        setLoggedOut(true);
      }
    }, /*#__PURE__*/_react.default.createElement(PageContent, {
      key: activeNavItem,
      title: activeNavItem
    })))
  );
};

var _default = ApplicationNavigationTest;
exports.default = _default;
