// terra-image/lib/terra-dev-site/doc/example/ImageDefault.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraImage = _interopRequireDefault(require("terra-image"));

var _x = _interopRequireDefault(require("../../assets/150x150.jpg"));

var _x2 = _interopRequireDefault(require("../../assets/1000x200.png"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ImageDefault = function ImageDefault() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h2", null, "Default"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x.default,
    alt: "default image"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h2", null, "Circle"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x.default,
    alt: "circle image",
    variant: "circle"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h2", null, "Rounded"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x.default,
    alt: "rounded image",
    variant: "rounded"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h2", null, "Thumbnail"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x.default,
    alt: "thumbnail image",
    variant: "thumbnail"
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h2", null, "Fluid"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x2.default,
    alt: "fluid image",
    isFluid: true
  }), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement("h2", null, "Placeholder"), /*#__PURE__*/_react.default.createElement("h3", null, "Invalid src prop (placeholder will show)"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: "/mock/failure/route",
    alt: "invalid src attribute image",
    placeholder: /*#__PURE__*/_react.default.createElement(_terraImage.default, {
      src: _x2.default,
      alt: "placeholder image"
    })
  }), /*#__PURE__*/_react.default.createElement("h3", null, "Valid src prop (src will show)"), /*#__PURE__*/_react.default.createElement(_terraImage.default, {
    src: _x.default,
    alt: "src image",
    placeholder: /*#__PURE__*/_react.default.createElement(_terraImage.default, {
      src: _x2.default,
      alt: "placeholder image"
    })
  }));
};

var _default = ImageDefault;
exports.default = _default;
